modules/api/src/main/java/net/countercraft/movecraft/TrackedLocation.java

```diff
@@ -0,0 +1,57 @@
+package net.countercraft.movecraft;
+
+import org.bukkit.World;
+import org.jetbrains.annotations.NotNull;
+
+public class TrackedLocation {
+    private int posX;
+    private int posY;
+    private int posZ;
+    private MovecraftLocation origin;
+
+    /**
+     * Creates a new TrackedLocation instance which tracks a location about an origin.
+     * @param origin The origin point at which the tracked location rotates about.
+     * @param location The absolute location to track.
+     */
+    public TrackedLocation(@NotNull MovecraftLocation origin, @NotNull MovecraftLocation location) {
+        this.origin = origin;
+        posX = location.getX() - origin.getX();
+        posY = location.getY() - origin.getY();
+        posZ = location.getZ() - origin.getZ();
+    }
+
+    /**
+     * Moves the origin point of the tracked location
+     */
+    public void translate(int dx, int dy, int dz) {
+        origin = new MovecraftLocation(origin.getX() + dx, origin.getY() + dy, origin.getZ() + dz);
+    }
+
+    /**
+     * Rotates the stored location about the origin point location.
+     * @param rotation A clockwise or counter-clockwise direction to rotate.
+     */
+    public void rotate(MovecraftRotation rotation) {
```

Comment:
The origin point of rotation needs to be passed into here by the `RotationTask`, since it will change a lot.